src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -0,0 +1,239 @@
+             const size_t dst_start,
+             const size_t count) {
+    // FIXME: Only 2 is supported now...
+    OPENVINO_ASSERT(dim == 2);
+    const auto* src_p = static_cast<uint8_t*>(src->data());
+          auto* dst_p = static_cast<uint8_t*>(dst->data());
+
+    const auto C  = src->get_shape()[1];
+    const auto SC = src->get_strides()[1];
+    const auto SH = src->get_strides()[dim];
+
+    for (int c = 0; c < C; ++c) {
+        const auto* sp = src_p + (C * SC) + (src_start * SH);
+              auto* dp = dst_p + (C * SC) + (dst_start * SH);
+        copy_rows(sp, count, SH, SH, dp);
+    }
+}
+
+ov::Tensor make_tensor_slice(ov::Tensor tensor, size_t dim, size_t start_pos, size_t end_pos) {
+    ov::Shape start_shape(std::vector<size_t>(tensor.get_shape().size(), 0u));
+    start_shape[dim] = start_pos;
+    ov::Shape end_shape = tensor.get_shape();
+    end_shape[dim] = end_pos;
+    return ov::Tensor(tensor, start_shape, end_shape);
+}
+
+ov::npuw::LLMInferRequest::LLMInferRequest(const std::shared_ptr<ov::npuw::LLMCompiledModel>& compiled_model,
+                                           const ov::npuw::LLMCompiledModel::KVCacheDesc& kvcache_desc)
+    : ov::ISyncInferRequest(compiled_model),
+      m_kvcache_desc(kvcache_desc) {
+    LOG_DEBUG("ov::npuw::LLMInferRequest::LLMInferRequest()");
+    LOG_BLOCK();
+    m_kvcache_request = compiled_model->kvcache_compiled->create_infer_request();
+    m_prefill_request = compiled_model->prefill_compiled->create_infer_request();
+
+    for (auto input_port : m_prefill_request->get_compiled_model()->inputs()) {
+        m_prefill_in_ports.emplace(input_port.get_any_name(), input_port);
+    }
+    for (auto output_port : m_prefill_request->get_compiled_model()->outputs()) {
+        m_prefill_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
+
+    for (auto input_port : m_kvcache_request->get_compiled_model()->inputs()) {
+        m_kvcache_in_ports.emplace(input_port.get_any_name(), input_port);
+    }
+    for (auto output_port : m_kvcache_request->get_compiled_model()->outputs()) {
+        m_kvcache_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
+    LOG_DEBUG("Done");
+}
+
+void ov::npuw::LLMInferRequest::prepare_for_new_conversation() {
+    // FIXME: for input_ids it must be padding from tokenizer that not available from here
+    // Get it from NPUW options
+    LOG_DEBUG("ov::npuw::LLMInferRequest::prepare_for_new_conversation()");
+    LOG_BLOCK();
+
+    auto prefill_compiled = m_prefill_request->get_compiled_model();
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("input_ids")), 0u);
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("attention_mask")), 0u);
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("position_ids")), 0u);
+    fill_tensor<int64_t>(m_kvcache_request->get_tensor(m_kvcache_in_ports.at("attention_mask")), 0u);
```

Comment:
not sure what's the reason to use this indirection here but ok.

Reply:
I think this is because of differences of used APIs, here we use `ov::SoPtr<ITensor>`, where we have only `get_tensor(...)` for ports

Reply:
Yes, I would like to explain my answer: we can't call `get_tensor("tensor_name")` from `ov::SoPtr<ITensor>` while we can do this with just `ov::Tensor`, so it seems like maps were introduced to have ability to get tensor by its name in a such way

Reply:
I get it, though it ain't much better than just having a bunch of named variables